### PR TITLE
`tls_client_auth`

### PR DESCRIPTION
**Pull request by paulswartz**
Fixes: #318 #327 

I think even this attempt with the profile is still a bit flaky. I need to figure out if the errors I'm seeing with the conformance suite are due to race conditions here, or something else.

### CODE REVIEW

**Comment by maennchen** [comment]
I was thinking to go more of a dynamic supervisor approach instead of the short keepalive one:

* Calculate [hash](https://www.erlang.org/doc/man/erlang#phash-2) of config
* Start profile for it if it does not yet exist
* Use profile
* Automatically stop profiles if not used for x time

**Comment by paulswartz** [comment]
Love that simpler option, and it works for the conformance tests.

File: src/oidcc_app.erl [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    try
+        inets:start(httpc, [{profile, httpc_profile()}])
+    catch
+        error:{already_started, _} -> ok
+    end,
+
+    % disable keep-alive
+    httpc:set_options(
+        [
+            {pipeline_timeout, 0},
```
Comment (maennchen) [inline src/oidcc_app.erl:41]: We can drop all of this with the new approach, right?

Comment (paulswartz) [reply src/oidcc_app.erl:41]: :+1: yep, forgot about that

**Review by maennchen — APPROVED** [review]

**Pull request merged**